### ISSUE TITLE
Store controller configs according to the UUID / "stable ID" instead of the OS controller order, and ignore configs for devices that aren't connected.

### ISSUE THREAD

**Issue by rb6502** [mamedev/mame#5835]
The current system of erasing the cfg if the user disconnects a controller was reasonable in 2000 when the DB-15 joystick port was dominant, shaky in 2010 when USB was dominant, and is completely unworkable in 2020 with Bluetooth and proprietary wireless dominant.

We have a system for configuring controllers by their "stable ID", but it requires extra work by the user and it really needs to be just how things work internally without the user needing to care.  Other emulators already do this.

In order to handle the old "disconnect the controller and there are no mappings for anything else" problem, we would not apply mappings for any stable ID controller that isn't present.

**Comment by galibert** [comment]
We can forcibly ensure that a mapping is available for the handful of
buttons needed to configure inputs.  That's, see, keyboard emulation, menu
popup, directions, quit and select, right?

**Comment by rb6502** [comment]
Right.  The idea is that if the connected devices don't cover mappings for those functions then we'd force the keyboard defaults for them.

**Comment by Firehawke** [comment]
This one will require considerable docs reworking, so I'm replying so that I'll remember to re-assess this once it's implemented and rework things as needed.
